src/Sulu/Bundle/DocumentManagerBundle/Command/PhpCRCleanerCommand.php

```diff
@@ -0,0 +1,244 @@
+    {
+        $this->output = $output;
+        $queryManager = $this->session->getWorkspace()->getQueryManager();
+        $rows = $queryManager->createQuery('SELECT * FROM [nt:unstructured]', 'JCR-SQL2')->execute();
+
+        $i = 0;
+        foreach ($rows->getNodes() as $node) {
+            $propertyData = $this->propertyParser->parse($node->getPropertiesValues('i18n:*'));
+
+            foreach ($propertyData as $locale => $contentData) {
+                /** @var array<Property> $contentData */
+                $output->writeln('======' . $locale . '======');
+
+                /** @var string $template */
+                $template = $contentData['template']->getValue();
+
+                try {
+                    $metaData = $this->structureMetaDataFactory->getStructureMetadata('page', $template);
+                } catch (StructureTypeNotFoundException|DocumentTypeNotFoundException) {
+                    $metaData = null;
+                }
+
+                if (null === $metaData) {
+                    $output->writeln(\sprintf(
+                        '[Skipping] Unable to load metaData for structure with type "page" and "%s"',
+                        $template
+                    ));
+                    continue;
+                }
+
+                $this->migrate('', $contentData, $metaData, 0);
+            }
+
+            $paths = $this->propertyParser->keyIterator($this->getNodesToDeleteAndClear());
+            foreach ($paths as $propertyPath) {
+                try {
+                    $node->getProperty($propertyPath)->remove();
+                    ++$i;
+                } catch (PathNotFoundException) {
+                    continue;
+                }
+            }
+        }
+        $this->session->save();
+
+        $this->output->writeln('Deleted properties: ' . $i);
+
+        return 0;
+    }
+
+    /**
+     * @param 16|32|64|128|256 $verbosity
+     */
+    private function depthDump(string $message, int $depth, int $verbosity = OutputInterface::VERBOSITY_NORMAL): void
+    {
+        $this->output->writeln(\str_repeat('  ', $depth) . $message, $verbosity);
+    }
+
+    /**
+     * @param array<mixed>|null $data
+     */
+    public function migrate(string|int $key, $data, object $metaData, int $depth): void
+    {
+        // If we are at the top of the metadata run we don't want to mess with that as it contains extra properties like seo and other extension data
+        if ($metaData instanceof StructureMetadata) {
+            foreach ($metaData->getProperties() as $subKey => $value) {
+                $this->depthDump('Iterating over: ' . $subKey, $depth, OutputInterface::VERBOSITY_VERY_VERBOSE);
+
+                /** @var array<mixed>|null $subValue */
+                $subValue = $data[$subKey] ?? null;
+                $this->migrate($subKey, $subValue, $value, $depth);
+            }
+
+            return;
+        }
+
+        if ($metaData instanceof BlockMetadata) {
```

Comment:
This section makes sense but only for Block and ComponentMetadata. But not really for PropertyMetadata right?